common-app-lib/src/cmr/common_app/services/search/query_to_elastic.clj

```diff
@@ -202,10 +202,27 @@
   (condition->elastic
    [{:keys [field query-str]} concept-type]
    (let [field (query-field->elastic-field field concept-type)]
-     {:query_string {:query (escape-query-string query-str)
-                     :analyzer :whitespace
-                     :default_field field
-                     :default_operator :and}}))
+     (cond
+       ;; for collection keyword search, we are using the new keyword2 index
+       ;; because we can not modify the existing index mapping.
+       (and (= :keyword-phrase field) (= :collection concept-type))
+         {:span_near
+           {:clauses [{:span_multi
+                        {:match
+                          {:wildcard
+                            {:keyword2 (escape-query-string query-str)}}}}]
+            :slop 0
+            :in_order true}}
+       (and (= :keyword field) (= :collection concept-type))
```

Comment:
No need to have this clause. We can use the `concept-type->field-mappings` function or something similar to map the keyword search field to the correct elasticsearch field and use the `else` clause handle anything that is not a keyword phrase.